Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -36,33 +36,567 @@ class Resources:
+            portal=endpoints.get('portal', defaults.endpoints.portal),
+            keyvault=endpoints.get('keyvault', defaults.endpoints.keyvault),
+        ),
+        suffixes=AzureCloudSuffixes(
+            storage_endpoint=suffixes.get('storage_endpoint', defaults.suffixes.storage_endpoint),
+            storage_sync_endpoint=suffixes.get('storage_sync_endpoint', defaults.suffixes.storage_sync_endpoint),
+            keyvault_dns=suffixes.get('keyvault_dns', defaults.suffixes.keyvault_dns),
+            mhsm_dns=suffixes.get('mhsm_dns', defaults.suffixes.mhsm_dns),
+            sql_server_hostname=suffixes.get('sql_server_hostname', defaults.suffixes.sql_server_hostname),
+            mysql_server_endpoint=suffixes.get('mysql_server_endpoint', defaults.suffixes.mysql_server_endpoint),
+            postgresql_server_endpoint=suffixes.get('postgresql_server_endpoint', defaults.suffixes.postgresql_server_endpoint),
+            mariadb_server_endpoint=suffixes.get('mariadb_server_endpoint', defaults.suffixes.mariadb_server_endpoint),
+            azure_datalake_store_file_system_endpoint=suffixes.get(
+                'azure_datalake_store_file_system_endpoint', defaults.suffixes.azure_datalake_store_file_system_endpoint),
+            azure_datalake_analytics_catalog_and_job_endpoint=suffixes.get(
+                'azure_datalake_analytics_catalog_and_job_endpoint',
+                defaults.suffixes.azure_datalake_analytics_catalog_and_job_endpoint),
+            acr_login_server_endpoint=suffixes.get('acr_login_server_endpoint', defaults.suffixes.acr_login_server_endpoint),
+            synapse_analytics_endpoint=suffixes.get('synapse_analytics_endpoint', defaults.suffixes.synapse_analytics_endpoint),
+            attestation_endpoint=suffixes.get('attestation_endpoint', defaults.suffixes.attestation_endpoint),
+        ))
+
+
+def url_trim_end_slash(url):
+    """
+        Will trim the end "/" from a URL.
+
+        Example:
+        "https://google.com/", "/"   => "https://google.com"
+        "https://google.com", "/"   => "https://google.com"
+
+        :type url: ``string``
+        :param url: URL string (required)
+
+        :return: trimmed url
+        :rtype: ``string``
+    """
+    return url.rstrip("/")
+
+
+def microsoft_defender_for_endpoint_get_base_url(is_gcc, params_endpoint_type, params_url):
+    # Backward compatible argument parsing, preserve the url and is_gcc functionality if provided, otherwise use endpoint_type.
+    if params_endpoint_type == MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE_CUSTOM or params_endpoint_type is None:
+        # When the integration was configured before our Azure Cloud support, the value will be None.
+        endpoint_type = "com"  # Default to "com"
+        if is_gcc is not None and is_gcc:  # Backward compatible.
+            endpoint_type = "gcc"
+            params_url = params_url or MICROSOFT_DEFENDER_FOR_ENDPOINT_API.get(endpoint_type)
+
+        if params_url is None:
```

Comment:
look at the exception description, it to prevent user input error if he leaves the server url field empty, by mistake. it should have a default value, but if the user leaves it blank it's a configuration error.